compiler/src/iree/compiler/Codegen/LLVMCPU/Utils.cpp

```diff
@@ -74,7 +74,8 @@ bool isLinalgGeneric2DTranspose(linalg::GenericOp genericOp) {
   // Check that the two indexing maps are a permutation of each other.
   SmallVector<AffineMap> indexingMaps = genericOp.getIndexingMapsArray();
   bool isTranspose =
-      indexingMaps[0].isPermutation() && indexingMaps[1].isIdentity();
+      (indexingMaps[0].isPermutation() && indexingMaps[1].isIdentity()) ||
+      (indexingMaps[1].isPermutation() && indexingMaps[0].isIdentity());
```

Comment:
this will be true if both the input and output indexing maps are transposed, does this need to be an xor?

Reply:
It won't be true if both the input and output are permutations. 
That'd be `(true && false) || (true && false)` => `false`.